src/input/type_record.cc

```diff
@@ -184,26 +191,11 @@ void Record::make_copy_keys(Record &origin) {
 }
 
 
-void Record::make_copy_keys_all() {
-	for(auto &ptr : data_->copy_from_ptr) {
-		ASSERT( ptr && TypeBase::was_constructed( ptr ), "Invalid pointer to source record for copy keys operation.\n");
-		Record tmp(*ptr);
-		make_copy_keys(tmp);
-		ptr = NULL;
-	}
-	data_->copy_from_ptr.clear();
-}
-
-
 
 Record &Record::derive_from(AbstractRecord &parent) {
```

Comment:
- if no key defined -> define key TYPE with empty shared ptr to actula type
- ASSERT that only TYPE key is defined (all other keys must com later)
- in finish check for TYP key and set proper selection from parent